app/client/genericState/actions.ts

```diff
@@ -0,0 +1,69 @@
+import { Body, Query } from "../api/client/types";
+import {
+  ActionGenerator,
+  APILifecycleActionTypesInterface,
+  CreateRequestAction,
+  LoadRequestAction,
+  ModifyRequestAction,
+  Registration
+} from "./types";
+
+class APILifeCycleActionType implements APILifecycleActionTypesInterface {
+  constructor(stage: string) {
+    this.stage = stage;
+  }
+
+  stage: string;
+  request = (name: string) => `${this.stage}_${name}_REQUEST`;
+  success = (name: string) => `${this.stage}_${name}_SUCCESS`;
+  failure = (name: string) => `${this.stage}_${name}_FAILURE`;
```

Comment:
I think this should enforce uppercase

<img width="714" alt="Screen Shot 2021-03-01 at 4 24 00 pm" src="https://user-images.githubusercontent.com/27039316/109455447-c1d62d00-7aaa-11eb-8e75-da5475534b7a.png">

Reply:
I could add a transform the the action string creators above?